internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -186,32 +201,40 @@ func (connector *SingleUseConnector) Connect(
 	// Write the prelogin packet back to the user
 	err = mssql.WritePreloginWithPacketType(clientBuffer, preloginResponse, mssql.PackReply)
 	if err != nil {
-		return connector.sendError("failed to write prelogin response: %s", err)
+		wrappedError := errors.Wrap(err, "failed to write prelogin response")
+		connector.sendError(wrappedError)
+		return nil, wrappedError
 	}
 
-	// Just like above, we don't need to concern ourselves with what the client would
-	// typically be sending to the server, since we are handling all communication with
-	// the server ourselves. We read the next packet so the client isn't blocked,
-	// while the driver handles the handshake.
-	err = clientBuffer.ReadNextPacket()
+	// We parse the client's Login packet so that we can pass on params to the server.
+	clientLogin, err := mssql.ReadLogin(clientBuffer)
 	if err != nil {
-		return connector.sendError("Failed to handle client prelogin request: %s", err)
+		wrappedError := errors.Wrap(err, "failed to handle client login")
+		connector.sendError(wrappedError)
+		return nil, wrappedError
 	}
 
-	// TODO: 	Send the login response to the client 	 (#1016)
-	// TODO: 	Verify appropriate errors are passed to the client (#1013)
-
-	// TODO: 	Replace this with an actual 'ok' message from the server
-	//			once login completes within the driver
-	if _, err = clientConn.Write(connector.CreateAuthenticationOKMessage()); err != nil {
-		return connector.sendError("Failed to send a successful authentication"+
-			" response to the client"+
-			": %s", err)
-	}
+	// Send the client login so that secretless can send honest client params to the server
+	clientLoginChannel <- *clientLogin
 
 	// Block continuation until driver has completed connection
 	<- connectPhaseFinished
+	if err != nil {
+		wrappedError := errors.Wrap(err, "failed to connect to mssql server")
+		connector.sendError(wrappedError)
+		return nil, wrappedError
+	}
```

Comment:
I don't see any way this err could have been set since line 215...

Reply:
@jonahx The go-routine that supplies a value to `connectPhaseFinished`. I left a comment about how this isn't a clear way of doing things.

Reply:
Oooooooh, yeah I see now.  I think we should definitely change this.  Easy fix:

`connectPhaseFinished` -> `connectResult`, and make it's type `chan error`, which can be nil or an error.

Reply:
@jonahx i thought to do that but the issue would still exist for driverConn. i think this is code that'll need reevaluating altogether.

Reply:
how about?

`chan ConnectResult` where `ConnectResult` is a struct that holds the `(driver, err)` pair?

Reply:
This is resolved in a following PR ->  https://github.com/cyberark/secretless-broker/pull/1050